File: SPSP/UI/spsp_mobile/lib/models/user_account.dart
import 'package:json_annotation/json_annotation.dart';

part 'user_account.g.dart';

@JsonSerializable()
class UserAccount {
  int id;
  String username;
  String email;
  String? password;
  String firstName;
  String lastName;
  bool valid;

  UserAccount(this.id, this.username, this.email, this.password, this.firstName,
      this.lastName, this.valid);

  /// A necessary factory constructor for creating a new User instance
  /// from a map. Pass the map to the generated `_$UserFromJson()` constructor.
  /// The constructor is named after the source class, in this case, User.
  factory UserAccount.fromJson(Map<String, dynamic> json) => _$UserAccountFromJson(json);

  /// `toJson` is the convention for a class to declare support for serialization
  /// to JSON. The implementation simply calls the private, generated
  /// helper method `_$UserToJson`.
  Map<String, dynamic> toJson() => _$UserAccountToJson(this);
}

File: SPSP/SPSP.Models/UserAccount/UserAccount.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

#nullable disable

namespace SPSP.Models
{
    public partial class UserAccount
    {
        public int Id { get; set; }
        public string Username { get; set; }
        public string Password { get; set; }
        public string Email { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public bool? Valid { get; set; }
        public virtual ICollection<UserAccountUserRole> UserAccountUserRoles { get; set; } = new List<UserAccountUserRole>();
    }
}
